### PR TITLE
first version terraform ecr, docker and makefile

### LINKED ISSUE #2: Create a GitHub repository
Create a project repository

### PR DESCRIPTION
**Pull request by CarineOliveira**
Next version plan: Make improvements to create variables and link to a WordPress image.

### CODE REVIEW

File: ecr/ecr.tf [hunk 1]
```diff
@@ -0,0 +1,8 @@
+resource "aws_ecr_repository" "project_ecr" {
+  name = "project_ecr"
```
Comment (caiocezart) [inline ecr/ecr.tf]: call it something like

name = "${var.project_name}-wordpress"

at the end it should be something like `group1-wordpress`

File: ecr/Makefile [hunk 2]
```diff
@@ -0,0 +1,31 @@
+all: init plan apply aws build tag push
```
Comment (caiocezart) [inline ecr/Makefile:6]: add a few variables here on the top:
ECR_REPO ?= ""
TAG ?= latest
IMAGE ?= $(ECR_REPO):$(TAG)

we can update those values in the future

File: ecr/Makefile [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+#Terraform
+
+init: 
+	terraform init
+
+plan: init
+	terraform plan
```
Comment (caiocezart) [inline ecr/Makefile]: change to `terraform plan -out=.terrafom-plan`

File: ecr/Makefile [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+#Terraform
+
+init: 
+	terraform init
+
+plan: init
+	terraform plan 
+
+apply: plan 
+	terraform apply -auto-approve
```
Comment (caiocezart) [inline ecr/Makefile]: I don't like the auto-approve.. we should always check the plan before applying, so:
- remove the `plan` linked with apply
- update to `terraform apply .terraform-plan` 

that way once you run `make plan` you get the results planned to change and if you do a `make apply` you won't need to type `yes`

File: ecr/Makefile [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+#AWS
+
+aws: apply
```
Comment (caiocezart) [inline ecr/Makefile]: remove the apply from here.. this build/tag/push part is just a test to see if the service works, let's make it independent

File: ecr/Makefile [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+plan: init
+	terraform plan 
+
+apply: plan 
+	terraform apply -auto-approve
+
+#AWS
+
+aws: apply 
+	aws ecr get-login-password --region ap-southeast-2 | docker login --username AWS --password-stdin 160372566358.dkr.ecr.ap-southeast-2.amazonaws.com/project_ecr
```
Comment (caiocezart) [inline ecr/Makefile]: update to `aws ecr get-login-password --region ap-southeast-2 | docker login --username AWS --password-stdin $(ECR_REPO)`

Comment (caiocezart) [reply ecr/Makefile]: that way we can easily update the ecr_repo URL changing the variable on the top of the file

File: ecr/Makefile [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+#Docker
+
+build: aws
```
Comment (caiocezart) [inline ecr/Makefile]: no need for `aws` here as mentioned above

File: ecr/Makefile [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+aws: apply 
+	aws ecr get-login-password --region ap-southeast-2 | docker login --username AWS --password-stdin 160372566358.dkr.ecr.ap-southeast-2.amazonaws.com/project_ecr
+
+#Docker
+
+build: aws  
+	docker build -t hello-world .
+
+tag: build
+	docker tag hello-world 160372566358.dkr.ecr.ap-southeast-2.amazonaws.com/project_ecr:latest
```
Comment (caiocezart) [inline ecr/Makefile]: no need for `build` here
also, update the command to `docker tag hello-world $(IMAGE)`

File: ecr/Makefile [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+push: tag
```
Comment (caiocezart) [inline ecr/Makefile]: update the push target to `push: tag aws` that way we will tag the image and auth to aws before trying to push

File: ecr/Makefile [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+	docker push 160372566358.dkr.ecr.ap-southeast-2.amazonaws.com/project_ecr:latest
```
Comment (caiocezart) [inline ecr/Makefile]: update to `docker push $(IMAGE)`

**Review by caiocezart — CHANGES_REQUESTED** [review]
very good work @CarineOliveira -- just a few notes based on best practices
- update your Makefile commands to use some variables so it make easier for other people to run on their own environments
- I suggest not using the `-auto-approve` with tf as it's always a good idea to check the plan before applying, I left a note on some other way that you could do it

**Comment by caiocezart** [comment]
Sorry forget one thing: if you go with my idea of exporting the `terraform plan -out=.terraform-plan` to a file, add to the `.gitignore` file on the repository root folder `.terraform*` so those files don't get added to git

File: ecr/ecr.tf (hunk 1 above)
Comment (CarineOliveira) [reply ecr/ecr.tf]: also included the terraform.tfvars file

File: ecr/Makefile (hunk 2 above)
Comment (CarineOliveira) [reply ecr/Makefile:6]: also included another variable IMAGE_NAME ?= "hello-world"

File: ecr/Makefile (hunk 3 above)
Comment (CarineOliveira) [reply ecr/Makefile]: also included the `-var-file terraform.tfvars` to refer to the variables file

**Comment by CarineOliveira** [comment]
> Sorry forget one thing: if you go with my idea of exporting the `terraform plan -out=.terraform-plan` to a file, add to the `.gitignore` file on the repository root folder `.terraform*` so those files don't get added to git

also added `.tfstate` to the `.gitignore` file

**Review by caiocezart — APPROVED** [review]
beautiful work 👏  feel free to merge your PR

**Comment by caiocezart** [comment]
this fix #13

**Pull request merged**